segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/BaseMetadataStore.java

```diff
@@ -143,13 +145,14 @@ public BaseMetadataStore() {
     /**
      * Begins a new transaction.
      *
+     * @param keysToLock Array of keys to lock for this transaction.
```

Comment:
You need to explain what is the behavior of this method with respect to locks. Is it going to block synchronously? Is it going to throw?

What if only one of the locks can't be acquired? Are the others going to be held while waiting for that one to be acquired?

Reply:
Updated.